### PR TITLE
OCPBUGS-14784: Honor global ingress configuration LoadBalancer type on AWS

### PR DESCRIPTION
**Pull request by csrwng**
**What this PR does / why we need it**:
The ingress.config.openshift.io API allows setting the type of LoadBalancer to use for AWS ingress. This commit honors that setting when creating the initial default IngressController for clusters on AWS.

**Which issue(s) this PR fixes** *(optional, use `fixes #<issue_number>(, fixes #<issue_number>, ...)` format, where issue_number might be a GitHub issue, or a Jira story*:
Fixes #OCPBUGS-14784

**Checklist**
- [x] Subject and description added to both, commit and PR.
- [x] Relevant issues have been referenced. 
- [x] This change includes unit tests.

### CODE REVIEW

File: support/globalconfig/ingress.go [hunk 1]
```diff
@@ -18,18 +18,22 @@ func IngressConfig() *configv1.Ingress {
...
+	}
 }
 
 func IngressDomain(hcp *hyperv1.HostedControlPlane) string {
 	if hcp.Spec.Configuration != nil && hcp.Spec.Configuration.Ingress != nil {
 		if len(hcp.Spec.Configuration.Ingress.AppsDomain) > 0 {
 			return hcp.Spec.Configuration.Ingress.AppsDomain
 		}
-		return hcp.Spec.Configuration.Ingress.Domain
+		if len(hcp.Spec.Configuration.Ingress.Domain) > 0 {
```
Comment (sjenning) [inline support/globalconfig/ingress.go:34]: This seems like an unrelated change where if `hcp.Spec.Configuration.Ingress != nil` but  `hcp.Spec.Configuration.Ingress.Domain` is not set, we default to the `apps.$baseDomain` rather than returning `""`.

**Comment by sjenning** [comment]
/lgtm
/override "Red Hat Trusted App Pipeline / hypershift-operator-main-on-pull-request"

File: support/globalconfig/ingress.go (hunk 1 above)
Comment (csrwng) [reply support/globalconfig/ingress.go:34]: It's related in that if you only specify only the NLB setting in config, we end up with a blank domain, unless you also specify it.
